### PR TITLE
OpenFGA Project Security Self-Assessment

### PR DESCRIPTION
**Pull request by gargabhay1999**
Created and added the first draft of the OpenFGA Project Security Self-Assessment.

### CODE REVIEW

**Comment by eddie-knight** [comment]
Hi there! I'm just getting started looking at your pull request, and I noticed the DCO check is failing.

You can look at the checks section of the PR (I believe it should always be below the last comment) and look for a red X highlighting the failed check. In this case, you can click `Details` for more information about how to get that check passing.
Screenshot 2023-12-08 at 8 35 18 AM

**Comment by eddie-knight** [comment]
For reviewers reference, OpenFGA is one of the few CNCF projects who have completed a self-assessment within the past couple months. 

It's very good to get it moved into this repo, but for reference the original doc lives here: https://github.com/openfga/openfga/blob/main/docs/security-self-assessment.md

**Comment by gargabhay1999** [comment]
> Hi there! I'm just getting started looking at your pull request, and I noticed the DCO check is failing.
> 
> You can look at the checks section of the PR (I believe it should always be below the last comment) and look for a red X highlighting the failed check. In this case, you can click `Details` for more information about how to get that check passing. Screenshot 2023-12-08 at 8 35 18 AM

Thanks for the review. I have resolved this fix.

File: assessments/projects/openfga/self-assessment.md [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+
+OpenFGA boasts exceptional speed in processing secure authorization check call. This swift authorization mechanism not only enhances efficiency but also reinforces the security posture, assuring robust protection for applications and platforms for diverse scales.
+
+OpenFGA provides a wide variety of SDK's, as well as easy integration for new SDK's. This reduces the chance of critical vulnerabilities due to compatibility issues.
+
+### Security Relevant
+
+Applications track and point to specific versions of authorization models.
+
+## Project Compliance
```
Comment (Rana-KV) [inline assessments/projects/openfga/self-assessment.md:188]: I'm putting my comments for future reference when the team is considering syncing with OpenFGA security self-assessment. I think this section can be more refined as PII might not be applicable in my sense. It is better to discuss with the team and finalize whether to include it or not. As the product level it should not be applicable in my understanding.

File: assessments/projects/openfga/self-assessment.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+## Self-Assessment Use
+
+This self-assessment is created by the OpenFGA team to perform an internal analysis of the project's security. It is not intended to provide a security audit of OpenFGA, or function as an independent assessment or attestation of OpenFGA's security health.
+
+This document serves to provide OpenFGA users with an initial understanding of OpenFGA's security, where to find existing security documentation, OpenFGA plans for security, and general overview of OpenFGA security practices, both for development of OpenFGA as well as security of OpenFGA.
+
+This document provides the CNCF TAG-Security with an initial understanding of OpenFGA to assist in a joint-assessment, necessary for projects under incubation. Taken together, this document and the joint-assessment serve as a cornerstone for if and when OpenFGA seeks graduation and is preparing for a security audit.
+
+## Security Functions and Features
```
Comment (Rana-KV) [inline assessments/projects/openfga/self-assessment.md:172]: I'm putting my comments for future reference when the team is considering syncing with OpenFGA security self-assessment. This section needs more details in the security function and features available for the project, the granularity and details are needed. This talks about projects role as security provider, but this section needs to talk about how the project is achieving itself security goals through the functions and features available.

File: assessments/projects/openfga/self-assessment.md [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+
+### Non-Goals
+
+* Tools for management of groups/roles/permissions not inherently provided to the end-users.
+* Does not intend to serve as a comprehensive data repository for non-authorization related data.
+* Does not aim to provide a complete authentication and Access Control Solution.
+
+## Self-Assessment Use
+
+This self-assessment is created by the OpenFGA team to perform an internal analysis of the project's security. It is not intended to provide a security audit of OpenFGA, or function as an independent assessment or attestation of OpenFGA's security health.
```
Comment (Rana-KV) [inline assessments/projects/openfga/self-assessment.md:166]: I'm putting my comments for future reference when the team is considering syncing with OpenFGA security self-assessment. If can specific who in the OpenFGA team is driving this topic along with the security pals.

File: assessments/projects/openfga/self-assessment.md [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+
+* Establish patterns and standards for externalized authorization.
+
+* Create architectural patterns, terminologies, and protocols that enable interoperability among different authorization systems.
+
+* Deliver an authorization service for any application component.
+
+* Enable centralized authorization decisions and permits diverse teams to implement authorization using a shared framework across various application components.
+
+### Non-Goals
```
Comment (Rana-KV) [inline assessments/projects/openfga/self-assessment.md:158]: I'm putting my comments for future reference when the team is considering syncing with OpenFGA security self-assessment. The non-goals and goals need to updated it is to generic. Some parts of goals and non-goals are overlapping.

**Comment by aaguiarz** [comment]
Hi @Rana-KV @eddie-knight @gargabhay1999 

I'm part of the OpenFGA team, we are working on an improved version of the self-assessment. Should we submit a PR to the https://github.com/AM-git-hub/tag-security/tree/main repository?

After we do that, would it be possible to schedule some time with someone from TAG-Security to review it?

Thanks!

**Comment by eddie-knight** [comment]
Absolutely, @aaguiarz! Thanks for raising that here. When you open the PR please tag us and we will make time to meet up to work through it.

@ragashreeshekar @JustinCappos  do we want to close this PR as duplicate in the meantime?

**Comment by JustinCappos** [comment]
The plan now is to clean up and then merge these into the repo in the meantime.  

The project team's modified assessment can replace and/or supplement the existing one, whenever it is ready.  However, we do want to have something in the repo for pretty much all of the projects which we can.

If you happen to complete your self assessment very quickly (before we merge this), then of course we will change the PR.  Don't feel under any rush to do so though!

File: assessments/projects/openfga/self-assessment.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
+# OpenFGA Security Self Assessment
+
+## Table of Contents
```
Comment (torinvdb) [inline assessments/projects/openfga/self-assessment.md:3]:
```suggestion
# OpenFGA Security Self Assessment

This assessment was created by community members as part of the [Security Pals](https://github.com/cncf/tag-security/issues/1102) process and is currently pending changes from the maintainer team.

## Table of Contents

```
@ragashreeshekar

File: assessments/projects/openfga/self-assessment.md [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+| Assessment Stage | Incomplete |
+| Software | https://github.com/openfga |
+| Security Provider | Yes. OpenFGA is used to decide if a subject (user, application) user can perform a specific action on a resource or not.|
+| Languages | Go, Java, Javascript, Python, C# |
+| SBOM | The Software Bill of Materials is not publicly available, but is included in each GitHub release using Syft, which is a CLI tool, and Go library for generating an SBOM from container images and filesystems. https://github.com/openfga/openfga/pull/683 |
+
+### Security Links
+
+  - https://github.com/openfga/openfga/security/policy
+  - https://github.com/orgs/openfga/security/risk
```
Comment (ragashreeshekar) [inline assessments/projects/openfga/self-assessment.md:49]:
```suggestion
| Doc | url |
| -- | -- |
| Security Policy | [OpenFGA Security Policy](https://github.com/openfga/openfga/security/policy) |
| Security Insights | [OpenFGA Security Insights](https://github.com/openfga/openfga/blob/main/SECURITY-INSIGHTS.yml.) |
| Security risks | [OpenFGA Security risks](https://github.com/orgs/openfga/security/risk) |
| -- | -- |
```

**Review by ragashreeshekar — APPROVED** [review]

**Comment by JustinCappos** [comment]
@ragashreeshekar Your DCO signoff is broken here!  Can you please fix so we can merge?

**Pull request merged**